ergo-lib/src/chain/json/hints.rs

```diff
@@ -0,0 +1,313 @@
+//! Hints JSON wrapper
+
+use crate::ergotree_interpreter::sigma_protocol::prover::hint::{
+    CommitmentHint, OwnCommitment, SimulatedCommitment,
+};
+use crate::ergotree_interpreter::sigma_protocol::{FirstProverMessage, ProverMessage};
+use crate::ergotree_ir::serialization::SigmaSerializable;
+use crate::ergotree_ir::sigma_protocol::sigma_boolean::SigmaProofOfKnowledgeTree;
+use ergotree_interpreter::sigma_protocol::dlog_protocol::FirstDlogProverMessage;
+use ergotree_interpreter::sigma_protocol::prover::hint::RealCommitment;
+use ergotree_interpreter::sigma_protocol::unproven_tree::NodePosition;
+use ergotree_ir::sigma_protocol::dlog_group::EcPoint;
+use ergotree_ir::sigma_protocol::sigma_boolean::ProveDlog as OtherProveDlog;
+use ergotree_ir::sigma_protocol::sigma_boolean::SigmaBoolean;
+use k256::elliptic_curve::generic_array::GenericArray;
+use k256::elliptic_curve::ops::Reduce;
+use k256::{Scalar, U256};
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+
+/// OwnCommitmentJson serialization
+#[derive(Serialize, Deserialize)]
+pub struct OwnCommitmentJson {
+    /// secret
+    pub secret: String,
```

Comment:
1. Why do you need these structs and their fields to be public?
2. Doc comments do not add any value. If they are internal types doc comments are not necessary.

Reply:
Wait, these `*CommitmentJson` structs are not used anywhere. Delete?

Reply:
Yes I delete that.